core/ssl.c

```diff
@@ -275,8 +275,8 @@ SSL_CTX *uwsgi_ssl_new_server_context(char *name, char *crt, char *key, char *ci
         }
 #if OPENSSL_VERSION_NUMBER >= 0x0090800fL
 #ifndef OPENSSL_NO_ECDH
-#ifdef NID_X9_62_prime256v1
-        EC_KEY *ecdh = EC_KEY_new_by_curve_name(NID_X9_62_prime256v1);
+#ifdef NID_secp384r1
```

Comment:
What about keeping support for both by moving this check before the other and transforming the other one in al elif ? I don't know if both are available for the same version of openssl so i think it would be better to avoid any eventual downgrading if that's the case.

Reply:
@xrmx 
Yes support for both is preferable and your idea would be the ideal solution for me. Considering I ran it like this on older Debian releases, this should support really old OpenSSL versions. I highly doubt you can realistically find old OpenSSL versions in use as Debian backports all security relevant releases.

Both Buster and Bullseye use openssl-1.1.1n, original was openssl-1.1.1d in 2019. Since there is only TLS1.2 and TLS1.3 now both are fully supported anyway, as TLS1.x is fully deprecated. What distros and python distributions did you have in mind with old openssl dependencies? Since Python 2.x is deprecated as well now.

I'm just a security freak by the way, using as strong crypto as software can possibly offer me. It felt sensible to use P384 key exchange and P384 signature schemes when both offered and supported by letsencrypt. In the wild you will find odd combinations like x25519 + RSA as on the Guardian's website.